### ISSUE TITLE
`--check-arg-defaults=True` appears unsatisfiable for un-annotated class attributes (enum members, `model_config`)

### ISSUE THREAD

**Issue by KelvinChung2000** [jsh9/pydoclint#298]
The following is an AI-generated summary of my problems. What is more important is the inconsistency I am seeing between `0.9.0` and `0.9.1`.  This is the related Dependabot update PR: https://github.com/FPGA-Research/FABulous/pull/918 

## Summary

When `--check-arg-defaults=True` is enabled, class attributes that are assigned **without a type annotation** (enum members, pydantic's `model_config`) seem to have no valid docstring form. The expected type hint that pydoclint computes for them begins with a comma, which cannot be expressed in a numpy docstring.

I would like to check whether I am holding this wrong before I work around it, since the workarounds all seem to cost more than the check is worth.

## Environment

- pydoclint 0.9.1
- docstring_parser_fork 0.0.16
- Python 3.12.12
- style: numpy

## Reproducer

```python
from dataclasses import dataclass, field
from enum import StrEnum


class Color(StrEnum):
    """An enum whose members are documented.

    Attributes
    ----------
    RED
        The color red.
    """

    RED = "RED"


@dataclass
class Config:
    """A dataclass whose docstring types are correct.

    Attributes
    ----------
    count : int
        A count.
    items : list[str]
        Some items.
    """

    count: int = 0
    items: list[str] = field(default_factory=list)
```

This file is **clean** under the default configuration:

```console
$ pydoclint --style=numpy repro.py
# no violations
```

But with `--check-arg-defaults=True`:

```console
$ pydoclint --style=numpy --check-arg-defaults=True repro.py
    5: DOC605: Class `Color`: Attribute names match, but type hints in these attributes do not match: RED
   18: DOC605: Class `Config`: Attribute names match, but type hints in these attributes do not match: count, items
```

## Issue 1: enum members have no passing form

`RED = "RED"` is an `ast.Assign` with no annotation, so in `Arg.fromArgWithMapping` the `typeHint` is `''` and the expected hint becomes:

```python
f'{arg.typeHint}, default={unparseName(defaultValue)}'   # -> ", default='RED'"
```

Asking pydoclint's own API confirms this:

```python
>>> Arg.fromArgWithMapping(Arg(name='RED', typeHint=''), mapping).typeHint
", default='RED'"
```

The expected string starts with a comma and has no type. To satisfy it a docstring would have to read:

```
RED : , default='RED'
```

which is not writable. I tried every form I could think of, and all three fail under `--check-arg-defaults=True`:

| Docstring | Result |
| --- | --- |
| `RED` (no type, as numpydoc renders enums) | DOC605 |
| `RED : str` | DOC605 |
| `RED : str, default='RED'` | DOC605 |

The same applies to pydantic's `model_config = ConfigDict(...)`, which is also an un-annotated assignment.

## Issue 2: dataclass fields require `default=field(...)`

For `Config` above, the only way I found to pass is:

```
count : int, default=0
items : list[str], default=field(default_factory=list)
```

`default=field(default_factory=list)` reads oddly, since the default value is an empty list, not a `field(...)` object. It is also the value of the `dataclasses.field()` sentinel rather than anything a caller could pass. Is documenting it this way the intent?

## Issue 3: the only workaround disables a useful check

The one configuration I found that lets function arguments use `, default=X` while enums still pass is:

```
--check-arg-defaults=True --check-class-attributes=False
```

But that turns off class-attribute checking entirely, which is the check we value most — it catches the common "the attribute type changed and the docstring was not updated" drift. Losing every DOC6xx check to gain argument-default checking is a bad trade for us.

## What we are actually trying to do

We want to follow numpydoc's rule for parameters, which says that `optional` is preferred only when the default would not be used as a value, and otherwise the default should be spelled out:

> If the default value would not be used as a value, `optional` is preferred.

So for `def f(indent: int = 0)` we would like to write `indent : int, default=0`, and we understand from `docs/style_deviations.md` that `, optional` is intentionally not accepted. That part is clear and we are happy to comply.

The difficulty is that `--check-arg-defaults` appears to be a single global switch that also applies to class attributes, where (a) some attributes cannot satisfy it at all and (b) the required spelling for dataclass fields is awkward.

## Questions

1. Is the un-annotated class attribute behaviour (Issue 1) a bug? It looks like `check-arg-defaults` should probably skip attributes that have no annotation, since there is no type to prefix the `, default=...` with.
2. Is `default=field(default_factory=list)` the intended documentation for a dataclass field, or should `check-arg-defaults` unwrap `dataclasses.field(...)` / pydantic `Field(...)`?
3. Is there a supported way to apply `--check-arg-defaults` to **function arguments only**, while keeping class-attribute checks (`--check-class-attributes=True`) in their current, non-default-checking form? If not, would you be open to that being separated?

**Comment by jsh9** [comment]
Hi @KelvinChung2000 , this AI-written descriptions seems to mix several problems together and it's written in a way that's difficult to understand.  (I read it twice and couldn't quite figure out what AI and/or you were trying to say.)

If there are separate problems, please open separate issues.

I'll close this issue for now.  Looking forward to clean human-written descriptions.
